vbfml/scripts/err_analysis

```diff
@@ -455,5 +489,313 @@ def compare_hist(ctx, input_dirs: str, number_epochs: int, save_name: str) -> No
     plt.savefig(pjoin(outdir, f"comparison_val_loss_{save_name}.pdf"))
 
 
+@cli.command()
+@click.pass_context
+@click.option(
+    "-i",
+    "--input-dir",
+    required=True,
+    help="path to the prediction directory of a model",
+)
+def roc(ctx, input_dir: str) -> None:
+    """
+    compute the roc curve of the model. /!/ input_dir point at the prediction_dir without a '/' at the end.
+    only compute and write in pkl file so it allows to use plot-roc on several model to compare in in one plot
+    """
+
+    # load model and prediction pickle file
+    loader = TrainingLoader(os.path.dirname(input_dir))
+    model = loader.get_model()
+    features, predictions, label_encoding, labels = load_cache(input_dir)
+    mjj = features["mjj"]
+    detajj = features["detajj"]
+
+    # compute ROC and precision recall curve
+    precision, recall, thresholds_prc = precision_recall_curve(labels, predictions)
+    fpr, tpr, thresholds_roc = roc_curve(labels, predictions)
+
+    # do the same for a simple cut on mjj and detajj
+    mjj_norm = mjj / mjj.max()
+    detajj_norm = detajj / detajj.max()
+    fpr_m, tpr_m, thresh_roc_m = roc_curve(labels, mjj_norm)
+    fpr_eta, tpr_eta, thresh_roc_eta = roc_curve(labels, detajj_norm)
+
+    # write data into pickle file
+    print("write data into pickle file")
+    data = {
+        "fpr": fpr,
+        "tpr": tpr,
+        "thresh_roc": thresholds_roc,
+        "precision": precision,
+        "recall": recall,
+        "thresholds_prc": thresholds_prc,
+        "fpr_m": fpr_m,
+        "tpr_m": tpr_m,
+        "mjj_max": mjj.max(),
+        "thresh_roc_m": thresh_roc_m,
+        "fpr_eta": fpr_eta,
+        "tpr_eta": tpr_eta,
+        "thresh_roc_eta": thresh_roc_eta,
+        "eta_max": detajj.max(),
+    }
+
+    with open(f"{input_dir}/roc.pkl", "wb") as f:
+        pickle.dump(data, f)
+
+
+def histogram(
+    data: pd.DataFrame,
+    variable: str,
+    cut: float,
+    outdir: str,
+) -> None:
+    plt.figure()
+    plt.hist(
+        data[data["labels"] == 0][variable],
+        bins=50,
+        density=True,
+        histtype="step",
+        label="background",
+    )
+    plt.hist(
+        data[data["labels"] == 1][variable],
+        bins=50,
+        density=True,
+        histtype="step",
+        label="signal",
+    )
+    plt.axvline(x=cut, color="k", linestyle="--", label=f"cut at {cut:.3f}")
+    plt.title(f"{variable}distribution")
```

Comment:
Nitpicking, could you leave a space between words here :) Something like:
```
plt.title(f"{variable} Distribution")
```

Reply:
done